File: src/main/java/com/codewars/kyu6/ConvertStringToCamelCase.java
package com.codewars.kyu6;

/**
 * @author Zurbaevi Nika
 */
public class ConvertStringToCamelCase {
    static String toCamelCase(String s) {
        String[] words = s.split("-|_");
        for (int i = 1; i < words.length; i++) {
            words[i] = words[i].substring(0, 1).toUpperCase() + words[i].substring(1);
        }
        return String.join("", words);
    }
}

File: src/main/kotlin/com/codewars/kyu6/ConvertStringToCamelCase.kt
package com.codewars.kyu6

/**
 * @author Zurbaevi Nika
 */
fun toCamelCase(str: String): String =
    str.split("-", "_")
        .mapIndexed { index, word ->
            if (index != 0) word.capitalize() else word
        }.joinToString("")
